client/lecturer/app/lecturer.factory.js

```diff
@@ -0,0 +1,79 @@
+(function() {
+    'use strict';
+
+    angular
+        .module('lecturer')
+        .factory('lecturerFactory', lecturerFactory);
+
+    /* @ngInject */
+    function lecturerFactory(lecturerService, $location) {
+        var username = '';
+        var observerCallbacks = [];
+        var Login = false;
+        
+        var service = {
+            login: login,
+            checkUserToken: checkUserToken,
+            logout: logout,
+            registerObserverCallback: registerObserverCallback
+        };
+
+        return service;
+
+        function login(credentials) {
+            if (!Login) {
+                Login = lecturerService.login;
+            }
+
+            var SHA256 = new Hashes.SHA256();
+            SHA256.setUTF8(true);
+            
+            // Prevent binding the hashed password into the input.
+            var form = {
+                email: credentials.email,
+                password: SHA256.hex(credentials.password)
+            };
+            
+            Login.save(form, loginSuccess, logoutSuccess);
```

Comment:
It's a bit confusing that logoutSuccess is called when login fails, so I think you should rename it.